docs/llm/reference.md

```diff
@@ -0,0 +1,147 @@
+
+Check out the [quickstart guide](quickstart.md) for a simple example that shows how to use this feature.
+
+## LLM Calculator
+As you can see in the quickstart above, big part of the configuration resides in `graph.pbtxt` file. That's because model server text generation servables are deployed as MediaPipe graphs with dedicated LLM calculator that works with latest [OpenVINO GenAI](https://github.com/ilya-lavrenov/openvino.genai/tree/ct-beam-search/text_generation/causal_lm/cpp/continuous_batching/library) solutions. The calculator is designed to run in cycles and return the chunks of reponses to the client.
+
+On the input it expects a HttpPayload struct passed by the Model Server frontend:
+```cpp
+struct HttpPayload {
+    std::string uri;
+    std::vector<std::pair<std::string, std::string>> headers;
+    std::string body;                 // always
+    rapidjson::Document* parsedJson;  // pre-parsed body             = null
+};
+```
+The input json content should be compatible with the [chat completions](./model_server_rest_api_chat.md) or [completions](./model_server_rest_api_completions.md) API.
+
+The input also includes a side packet with a reference to `LLM_NODE_RESOURCES` which is a shared object representing an LLM engine. It loads the model, runs the generation cycles and reports the generated results to the LLM calculator via a generation handler. 
+
+**Every node based on LLM Calculator MUST have exactly that specification of this side packet:**
+
+`input_side_packet: "LLM_NODE_RESOURCES:llm"`
+
+**If it's modified, model server will fail to provide graph with the model**
+
+On the output the calculator creates an std::string with the json content, which is returned to the client as one response or in chunks with streaming.
+
+Let's have a look at the graph from the graph configuration from the quickstart:
+```protobuf
+input_stream: "HTTP_REQUEST_PAYLOAD:input"
+output_stream: "HTTP_RESPONSE_PAYLOAD:output"
+
+node: {
+  name: "LLMExecutor"
+  calculator: "HttpLLMCalculator"
+  input_stream: "LOOPBACK:loopback"
+  input_stream: "HTTP_REQUEST_PAYLOAD:input"
+  input_side_packet: "LLM_NODE_RESOURCES:llm"
+  output_stream: "LOOPBACK:loopback"
+  output_stream: "HTTP_RESPONSE_PAYLOAD:output"
+  input_stream_info: {
+    tag_index: 'LOOPBACK:0',
+    back_edge: true
+  }
+  node_options: {
+      [type.googleapis.com / mediapipe.LLMCalculatorOptions]: {
+          models_path: "./"
+      }
+  }
+  input_stream_handler {
+    input_stream_handler: "SyncSetInputStreamHandler",
+    options {
+      [mediapipe.SyncSetInputStreamHandlerOptions.ext] {
+        sync_set {
+          tag_index: "LOOPBACK:0"
+        }
+      }
+    }
+  }
+}
+```
+
+Above node configuration should be used as a template since user is not expected to change most of it's content. Fields that can be safely changed are:
+ - `name`
+ - `input_stream: "HTTP_REQUEST_PAYLOAD:input"` - in case you want to change input name
```

Comment:
why would it be changed?

Reply:
I'm just listing parameters that can be changed without affecting functionality. 
There's no particular reason to change this one, but if someone wants to go with for example "request" instead of "input" they can.

Reply:
I see but I think we are complicating things more than needed here. Users should just change the node options.